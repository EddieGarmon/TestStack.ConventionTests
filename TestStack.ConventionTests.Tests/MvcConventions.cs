namespace TestStack.ConventionTests.Tests
{
    using NUnit.Framework;
    using Shouldly;
    using TestAssembly.Controllers;
    using TestStack.ConventionTests.ConventionData;
    using TestStack.ConventionTests.Conventions;

    [TestFixture]
    public class MvcConventions
    {
        [Test]
        public void controller_conventions()
        {
            var types = Types.InAssemblyOf<TestController>();
            var convention = new MvcControllerNameAndBaseClassConvention();

            var failures = Convention.GetFailures(convention, types);

            failures.ShouldMatchApproved();
        }

        [Test]
        public void api_controller_conventions()
        {
            var types = Types.InAssemblyOf<TestController>();
            var convention = new ApiControllerNamingAndBaseClassConvention();

            var failures = Convention.GetFailures(convention, types);

            failures.ShouldMatchApproved();
        } 
    }
}