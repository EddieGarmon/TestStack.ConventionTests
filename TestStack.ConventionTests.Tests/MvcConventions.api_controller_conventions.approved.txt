Failed: 'Api Controllers must be suffixed with Controller' for 'TestAssembly'
-----------------------------------------------------------------------------

	TestAssembly.Controllers.BarApiControler


Failed: 'Types named *Controller must inherit from ApiController or Controller' for 'TestAssembly'
--------------------------------------------------------------------------------------------------

	TestAssembly.Controllers.TestApiController
	TestAssembly.Controllers.TestController